test/e2e/main_test.go

```diff
@@ -366,6 +366,29 @@ func TestManualMeteringInstall(t *testing.T) {
 			},
 			MeteringConfigManifestFilename: "nfs.yaml",
 		},
+		{
+			Name:                      "ReportExpiration",
+			MeteringOperatorImageRepo: meteringOperatorImageRepo,
+			MeteringOperatorImageTag:  meteringOperatorImageTag,
+			Skip:                      false,
+			InstallSubTests: []InstallTestCase{
+				{
+					Name:         "testReportIsDeletedWhenNoDeps",
+					TestFunc:     testReportIsDeletedWhenNoDeps,
+					ExtraEnvVars: []string{},
+				},
+				{
+					Name:         "testReportIsNotDeletedWhenReportDependsOnIt",
+					TestFunc:     testReportIsNotDeletedWhenReportDependsOnIt,
+					ExtraEnvVars: []string{},
+				},
+				//{
+				//	Name:     "testReportIsNotDeletedWhenReportQueryDependsOnIt",
+				//	TestFunc: testReportIsNotDeletedWhenReportQueryDependsOnIt,
+				//},
+			},
+			MeteringConfigManifestFilename: "prometheus-metrics-importer-enabled.yaml",
+		},
```

Comment:
Once we get closer to merging this, I don't think this kind of testing coverage requires it's own dedicated test installation. Should we deflate those sub-tests into existing test installations?

Reply:
agreed, I'll move them into another top level test.

Reply:
Going with sticking them under: `ValidHDFS-ReportStaticInputData` 

FWIW: None of the top-level test "suite" names make much sense to me.

RANT: Go's test infra is primitive, making coders look through a long struct for tests and subtests is bogus.

Reply:
Yeah, agreed on the naming of tests doesn't make much sense. I'm under the impression that we were going to have tests that caught some of the common static validation errors we were seeing (e.g. `spec.storage` wasn't properly configured) but never got to that part in the e2e suite. I would also expect that kind of validation to live in its own top-level tests, so we can rename any of the `Valid*-<Storage>-...` tests.

But for the go test infra part, most of the test code is custom logic and we only utilize a portion of the stdlib testing suite, and the recommendation to use table driven tests.